packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -245,7 +250,18 @@ class AndroidStudio implements Comparable<AndroidStudio> {
     }
     AndroidStudio? newest;
     for (final AndroidStudio studio in studios.where((AndroidStudio s) => s.isValid)) {
-      if (newest == null || studio.compareTo(newest) > 0) {
+      if (newest == null) {
+        newest = studio;
+        continue;
+      }
+
+      if (studio.version != null && newest.version == null) {
+        newest = studio;
+      } else if (studio.version == null && newest.version == null &&
+            studio.directory.compareTo(newest.directory) > 0) {
```

Comment:
```suggestion
            studio.directory > newest.directory) {
```

Reply:
Actually, yeesh, so we're just alphabetically comparing these strings?

Reply:
And I see even though string has a `compareTo`, it doesn't implement comparable

Reply:
Let me file a tracking issue about investigating whether we should really be doing this.

Reply:
Can you add a TODO with a link to https://github.com/flutter/flutter/issues/124880

Reply:
> Actually, yeesh, so we're just alphabetically comparing these strings?

Only if 1) both installations have equivalent versions or 2) neither installation have known versions. I wonder what the original inspiration for this was...perhaps it was just a way keep ordering consistent even if the discovery logic changed.

Reply:
Actually, I might be slightly incorrect here. Let me re-examine the original code.

Reply:
Actually, reading this more, I can see this is a last-resort fallback.

Reply:
I think I have the logic right. I added another test